pkg/volume/git_repo/git_repo.go

```diff
@@ -169,6 +171,10 @@ func (b *gitRepoVolumeMounter) GetAttributes() volume.Attributes {
 
 // SetUp creates new directory and clones a git repo.
 func (b *gitRepoVolumeMounter) SetUp(mounterArgs volume.MounterArgs) error {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.GitRepoVolumeDriver) {
+		return fmt.Errorf("git-repo volume plugin has been disabled")
```

Comment:
Not sure if it makes sense to be more explicit in the error: `git-repo volume plugin has been disabled; if necessary, it may be re-enabled by k8s operator in current release, but please note, it will be completely removed in a future release`?

Reply:
I wouldn't start by asserting future removal yet, the first step is just default disablement and conditional re-enablement

This could include the feature gate used to disable the feature in the error if that would help an admin trying to find details about the disablement and / or escape hatch re-enablement

Reply:
Updated to say that this was default disabled and can be re-enabled by enabling the feature-gate `GitRepoVolumeDriver`